bindings/java/c/host.c

```diff
@@ -40,31 +40,28 @@ static bool account_exists_fn(struct evmc_host_context* context, const evmc_addr
     const char java_method_signature[] = "(I[B)I";
     assert(context != NULL);
     JNIEnv* jenv = attach();
-    if (jenv != NULL)
```

Comment:
Why did you remove these?

Reply:
cmake fails to compile with this. Also, the behavior is surprising - if JNI fails to attach, we should fail, not avoid executing the operation.

Reply:
Can you show an error log?

Reply:
```/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:370:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:402:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:370:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:365:5: note: variable 'result' is declared here
    struct evmc_result result;
    ^
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:412:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:439:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:412:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:407:5: note: variable 'result' is declared here
    struct evmc_tx_context result;
    ^
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:450:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:481:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:450:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:445:5: note: variable 'result' is declared here```

Reply:
I think the preference would be to individually fix these issues, in separate PRs. I agree this extra conditional isn't the nicest, and probably we have multiple options:
1) Move the declarations around so the compiler won't complain
2) Instead of the conditional, throw a Java exception?
3) Instead of conditional, add an assertion (does an abort).